var/spack/repos/builtin/packages/libibumad/package.py

```diff
@@ -0,0 +1,27 @@
+
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Libibumad(CMakePackage):
+    """This package installed the user-spack libraries and headers for libibumad.
+       This is a subset of the linux-rdma distribution."""
+
+    homepage = "https://github.com/linux-rdma/"
+    url      = "https://github.com/linux-rdma/rdma-core/archive/v25.0.tar.gz"
+
+    version('25.0', sha256='d735bd091d13e8a68ce650e432b5bdc934fc7f1d5fb42a6045278a5b3f7fe48b')
+
+    depends_on('libnl')
+
+    def build(self, spec, prefix):
+        with working_dir(join_path(self.build_directory, 'libibumad')):
+            make()
```

Comment:
I think if you set `build_directory` at the class level this will happen automatically for you, like:

```
depends_on('libnl')

build_directory = 'libibumad'
```

The `CMakePackage` class this inherits from should be able to work with that since this is the typical `make`/`make install` pattern.

Reply:
OK thanks. I'll try this.

Reply:
I looked into this, and it will not work because cmake needs to be executed from one level up from the libibumad sub-directory.  This is because libibumad is a sub-project of linux-rdma.